django/db/models/operators.py

```diff
@@ -0,0 +1,52 @@
+from django.core.exceptions import FieldError
+from django.db.models.expressions import CombinedExpression, DurationValue
+from django.db.models.fields import (
+    CharField, DateField, DateTimeField, DurationField, TextField, TimeField,
+)
+from django.db.models.functions import ConcatPair
+
+__all__ = ()
+
+
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateField, DateField, commutative=False)
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateTimeField, DateTimeField, commutative=False)
+@CombinedExpression.dispatch(CombinedExpression.SUB, TimeField, TimeField, commutative=False)
+class TemporalSubtraction(CombinedExpression):
+    def __init__(self, lhs, connector, rhs, output_field):
+        super(TemporalSubtraction, self).__init__(lhs, self.SUB, rhs, output_field=DurationField())
```

Comment:
@charettes unfortunately I can't instantiate a `Field` at the module level (I get `ImproperlyConfigured` as visible in Jenkins), so I've changed the invocation to `output_field=output_field or DurationField()`